### DISCUSSION TITLE
iTunes API

### DISCUSSION THREAD

**Discussion by gAlleb** [AzuraCast/AzuraCast#6859] [General]
Just wondering. Why not iTunes API?  Or am I dreaming and it's already a feature? 

Last.fm is very very poor on cover art. About 20% of my music has no cover art and last.fm recognizes about 0,01% of that music while iTunes API recognizes 95%. 

It is quite easy to write. As for now I use on my home radio page XHR requests to iTunes API to be sure I get the cover art. 

Could we somehow incorporate iTunes API?

P.S. I don’t know any reason not to use this option for now, if it exists, please let me know so that I don’t introduce more stupid things in the future.

**Comment by Vaalyn** [comment]
From the iTunes search API terms:

> Developers may use promotional content in the API, including previews of songs, music videos, album art and App icons only to promote store content and not for entertainment purposes.

See: https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/iTuneSearchAPI/index.html

So in order to be legally allowed to incorporate the API it would also be necessary to add backlinks to the respective title on the iTunes store.

Not sure if there are other legal requirements but that is something to keep in mind.

**Reply by gAlleb** [reply]
Oh! I see now. No more stupid things in the future.
Thanks!
